Xamarin.Essentials/Types/Location.shared.cs

```diff
@@ -104,9 +112,11 @@ public Location(Location point)
 
         public override string ToString() =>
             $"{nameof(Latitude)}: {Latitude}, {nameof(Longitude)}: {Longitude}, " +
-            $"{nameof(Altitude)}: {Altitude ?? 0}, {nameof(Accuracy)}: {Accuracy ?? 0}, " +
-            $"{nameof(VerticalAccuracy)}: {VerticalAccuracy ?? 0}, " +
-            $"{nameof(Speed)}: {Speed ?? 0}, {nameof(Course)}: {Course ?? 0}, " +
+            Altitude != null ? $"{nameof(Altitude)}: {Altitude}, " : string.Empty +
```

Comment:
Here you would just do:

```
$"{nameof(Altitude)}: {Altitude ?? string.Emptry}
```

Reply:
I still think it should output numbers though as the idea is to print everything out.

Reply:
I really don't think outputting zeros for non-existing information is a good idea. For the accuracies it's not so much of a problem, since those are usually positive. But the values for altitude, speed and course can easily become zero in real-world applications, so how do you distinguish an actual zero value from a `null` value, if `ToString` outputs a zero for both?

If we leave the value blank, then I don't see much reason to print out the property name either. Often a location will not include the speed and course information, in particular if the variable really just specifies a 'location' in the actual sense (i.e. without movement). In that case it can be a bit annoying if those property names are printed all the time, in particular if you print out lots of locations.

Reply:
We must determine the reason for the existence of the `ToString`. I would go and say that if it is null, then just print `Accuracy=,` This is purely for devs, so the zero was wrong, but excluding it is also not correct.

Reply:
> Here you would just do:
> 
> ```
> $"{nameof(Altitude)}: {Altitude ?? string.Emptry}
> ```

This actually does not work:

```
Error CS0019: Operator '??' cannot be applied to operands of type 'double?' and 'string' (CS0019)
```

Reply:
> I would go and say that if it is null, then just print `Accuracy=,` This is purely for devs, so the zero was wrong, but excluding it is also not correct.

Alright then, I'm now using the following pattern for the nullable data members:

```
$"{nameof(Altitude)}: " + Altitude != null ? $"{Altitude}" : string.Empty + ", " +
```